soundfile.py

```diff
@@ -167,6 +167,33 @@
     __libsndfile_version__ = __libsndfile_version__[len('libsndfile-'):]
 
 
+class SoundFileError(RuntimeError):
+    """Base class for all SoundFile-specific errors.
+
+    :ivar code: libsndfile internal error number or `None`
+    :ivar error_string: raw libsndfile error message or `None`
+    """
+    def __init__(self, message, code=None, error_string=None):
+        self.args = (message, code, error_string)
+        self.message = message
+        self.code = code
+        self.error_string = error_string
+    def __str__(self):
+        return self.message
+
+
+def _sndfile_error(code, message=None, prefix=""):
+    """Return one of the SoundFileError exception, proper for the given
+    sndfile error code.
+
+    Use provided message or auto-generate one with sf_strerror().
+    """
+    err_str = _snd.sf_error_number(code)
+    if message is None:
+        message = prefix + _ffi.string(err_str).decode('utf-8', 'replace')
```

Comment:
I think it's better to store the separate components in the error class, and concatenate the final message in `__str__`.

Reply:
You are right. I was focused on keeping as much of the current behaviour as possible.
As 'args' of the exception are changed anyway, I can move the prefix there too. The exception class will look even more artificial then, though (for backward compatibility – parsing exception messages used be the only way to differentiate errors and is even unit-tested in the package).

Reply:
Yeah, I don't care very much for `args` but it would be good to keep the final messages unless there is a very good reason to change them.